test/request-error.test.js

```diff
@@ -412,3 +415,89 @@ test('test request.routeOptions.version', t => {
     })
   })
 })
+
+test('customErrorHandler should throw for json err and stream response', t => {
+  t.plan(7)
+
+  const logStream = split(JSON.parse)
+  const fastify = Fastify({
+    logger: {
+      stream: logStream,
+      level: 'error'
+    }
+  })
+
+  t.teardown(fastify.close.bind(fastify))
+
+  fastify.get('/', async (req, reply) => {
+    const stream = new Readable({
+      read () {
+        this.push('hello')
+      }
+    })
+    process.nextTick(() => stream.destroy(new Error('stream error')))
+
+    reply.type('application/text')
+    await reply.send(stream)
+  })
+
+  fastify.setErrorHandler((err, req, reply) => {
+    t.equal(err.message, 'stream error')
+    reply.code(400)
+    reply.send({ error: err.message })
+  })
+
+  logStream.once('data', line => {
+    t.equal(line.msg, 'Attempted to send payload of invalid type \'object\'. Expected a string or Buffer.')
+    t.equal(line.level, 50)
+  })
+
+  fastify.listen({ port: 0 }, err => {
+    t.error(err)
+
+    sget({
+      method: 'GET',
+      url: getServerUrl(fastify) + '/'
+    }, async (err, res) => {
+      t.error(err)
+      t.equal(res.statusCode, 500)
+      t.equal(res.payload, undefined)
+    })
+  })
+})
+
+test('customErrorHandler should not call handler for in-stream error', t => {
```

Comment:
I do understand this case, since you have no chance to reply when the socket is hang.